### ISSUE TITLE
Updating packages when using pnpm workspaces

### ISSUE THREAD

**Issue by tvvignesh** [pnpm/pnpm#2982]
### Reproduction:

Hi. I use pnpm workspaces in all my projects and it works great. Today, I noticed an issue where I was not able to do a `pnpm update` since it fails with workspace packages.

For eg. I had this:

<img width="409" alt="1" src="https://user-images.githubusercontent.com/1165845/99267062-783e3b00-2849-11eb-88b1-88c23b88ddeb.png">

where `@timecampus/core` is a package within my workspace (I use calver for versioning) and I want to update every other package in my `package.json` except my local workspace packages. But I get the error as seen in the pic above. Both recursive and isolated updates fail

### Expected behavior:

pnpm updates every other package ignoring my local workspace packages.

### Actual behavior:

pnpm fails to update deps

### Additional information:

 - `node -v` prints: v15.1.0
 - Windows, macOS, or Linux?: Ubuntu 20.04

**Comment by tvvignesh** [comment]
@zkochan I guess I understood what the issue is. If you see the pic below:

<img width="632" alt="1" src="https://user-images.githubusercontent.com/1165845/99274727-ab37fd00-2850-11eb-8006-ab6f23cf1e93.png">

I did not have the prefix `workspace:` in my workspace packages which was added automatically when I did `pnpm -r up` for the first time. Now, because of this, pnpm works well for the first time.

If you try running `pnpm -r up` once more after this (with `workspace:` prefix still on the packages), update fails with the error I mentioned above.

So, in summary, update fails when version number has the prefix `workspace:` for workspace packages.

**Comment by zkochan** [comment]
What is the version of `@timecampus/core` in the monorepo?

**Comment by tvvignesh** [comment]
@zkochan Hi. This is what the version of `@timecampus/core` is in the monorepo:

```json
{
  "name": "@timecampus/core",
  "version": "2020.12.30-01",
}
```

**Comment by tvvignesh** [comment]
I also noticed that pnpm checks in the registry during `pnpm up` if the `--latest` flag is added but `pnpm up` without the latest flag checks locally first as seen in the image below. I also tried adding the `prefer-workspace-packages` as documented here: https://pnpm.js.org/en/workspaces#prefer-workspace-packages and still got the same result. Btw, I did upgrade to **5.13.1** from the above mentioned pnpm version.

<img width="367" alt="1" src="https://user-images.githubusercontent.com/1165845/99560455-e3c00e00-29eb-11eb-962f-a8eb7ea53f19.png">

**Comment by zkochan** [comment]
It seems like you did not create a `pnpm-workspace.yaml` file in the root of your monorepo. You need to create it: https://pnpm.js.org/en/pnpm-workspace_yaml

**Comment by tvvignesh** [comment]
@zkochan I do have a `pnpm-workspace` and have been using it for close to two months and it has been working great 

<img width="196" alt="1" src="https://user-images.githubusercontent.com/1165845/99579427-2e4d8480-2a04-11eb-91fc-e9af61a00ff7.png">

This problem does not occur for installs. For instance (As the message says, it is going through the workspace packages):

<img width="729" alt="2" src="https://user-images.githubusercontent.com/1165845/99579543-5210ca80-2a04-11eb-9b81-cff44a5710be.PNG">

**Comment by tvvignesh** [comment]
@zkochan Hi. Just wanted to let you know that this problem still occurs. As of now I am replacing the `workspace:` prefix which gets added everytime and rerun the `pnpm -r up` everytime.

<img width="833" alt="3" src="https://user-images.githubusercontent.com/1165845/102054093-dc372d80-3e0e-11eb-86a1-6da7eed7a586.PNG">

**Comment by tvvignesh** [comment]
And if you add the `workspace:` prefix to all packages, this happens

<img width="733" alt="1" src="https://user-images.githubusercontent.com/1165845/102980686-f1f2d400-452d-11eb-88a0-cc5386405753.PNG">

**Comment by JobinJia** [comment]
i have same proplem.
if `package.json` field `version` value is not  `1.0.0` in sub package
when i run command with `pnpm install`  or `pnpm upgrade`,
the console will print this tips 

![image](https://user-images.githubusercontent.com/9244703/149060657-682dcf6d-c67c-4208-8e50-ff2a005c66e0.png)

**Comment by lllopo** [comment]
I also have an issue, that seems related to me, but if it is not - feel free to move it to a separate one. I have an app centric monorepo, so structure like :

```
root
|--- apps
|        |----- my app
|
|--- packages
         |----- pkg1
         |----- pkg2
```
I have `auto-install-peers` = `true`.
When I do `pnpm install` on the root (no package.json in the root), the command runs fine, everything is installed properly and builds properly. 
If I try `pnpm update`, though, it complains about missing package.json. 
If I try `pnpm update -r` a real big mess happens - the command runs, BUT all my peer dependencies in my package.json get copied in the dependencies section (which I definitely don't want). 
If I turn off `auto-install-peers`, then `pnpm update -r` runs fine, without messing with my package.json files, BUT the peers are missing (of course) and the projects fail to build.

So I see two issues here :
1. `pnpm update` does not understand the concept of a workspace the way `pnpm install` does.
2. `pnpm update -r` physically changes the package.json files without consent (!!!)

If you think this is not a bug, but it is due to some misconfiguration from my side - a helpful hand would be appreciated.

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1: `pnpm -r update` in a workspace whose projects depend on each other over `workspace:*` completes without error.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
